frontend/src/bundles/common/components/base-layout/components/suggestion-for-subscribing.tsx

```diff
@@ -0,0 +1,85 @@
+import { differenceInDays } from 'date-fns';
+
+import { ComponentSize } from '~/bundles/common/enums/component-size.enum.js';
+import { type SubscriptionGetItemResponseDto } from '~/bundles/common/enums/enums.js';
+import { AppRoute } from '~/bundles/common/enums/enums.js';
+import {
+    useAppDispatch,
+    useEffect,
+    useNavigate,
+    useState,
+} from '~/bundles/common/hooks/hooks.js';
+import { actions as subscriptionActions } from '~/bundles/subscription/store/subscriptions.js';
+
+import { useAppSelector, useCallback } from '../../../hooks/hooks.js';
+import { Button, ButtonVariant, Modal } from '../../components.js';
+
+const SuggestionForSubscribing: React.FC = () => {
+    const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const { currentSubscription } = useAppSelector(({ subscriptions }) => {
+        return {
+            currentSubscription:
+                subscriptions?.currentSubscription as SubscriptionGetItemResponseDto,
+        };
+    });
+
+    const handleCloseModal = useCallback((): void => {
+        localStorage.setItem('lastSuggestionTime', String(Date.now()));
+        setIsModalOpen(false);
+    }, [setIsModalOpen]);
+
+    const handleNavigateToSubscriptionPage = useCallback((): void => {
+        localStorage.setItem('lastSuggestionTime', String(Date.now()));
+        navigate(AppRoute.PROFILE_SUBSCRIPTION);
+        setIsModalOpen(false);
+    }, [navigate, setIsModalOpen]);
+
+    const handleLoadCurrentSubscription = useCallback((): void => {
+        void dispatch(subscriptionActions.loadCurrentSubscription());
+    }, [dispatch]);
+
+    useEffect(() => {
+        handleLoadCurrentSubscription();
+    }, [handleLoadCurrentSubscription]);
+
+    useEffect(() => {
+        const lastSuggestionTimeString =
+            localStorage.getItem('lastSuggestionTime');
+        const lastSuggestionTime = lastSuggestionTimeString
+            ? Number.parseInt(lastSuggestionTimeString, 10)
+            : null;
+        const numberOfDaysAgo = lastSuggestionTime
+            ? differenceInDays(new Date(), lastSuggestionTime)
+            : 0;
```

Comment:
```suggestion
        const lastSuggestionTime = Number.parseInt(lastSuggestionTimeString, 10) || Date.now();
        const numberOfDaysAgo = differenceInDays(new Date(), lastSuggestionTime);
```

Can we check it like this? Also add 'lastSuggestionTime' to enum for local storage  keys.

Reply:
thanks, fixed.